src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,129 @@
+import * as classNames from 'classnames';
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import Button from '../Button/Button';
+import * as Rivet from '../util/Rivet';
+import ModalEvent from './ModalEvent';
+
+
+export interface ModalProps {
+    /**
+     * Determines whether the modal is shown or not
+     */
+    isOpen?: boolean;
+    /**
+     * Optional function to call to raise when the alert is dismissed. If undefined, the modal will act as a dialog (i.e., no close button, 
+     * will not close on outside click or escape key)
+     */
+    onDismiss?: Rivet.Action;
+    /**
+     * The content of the modal's header
+     */
+    title: React.ReactNode | string;
+}
+
+const ModalCloseButton = ({ onDismiss }) => (
+    <Button className="rvt-modal__close" data-modal-close="close" onClick={onDismiss}>
+        <span className="rvt-sr-only">Close</span>
+        <svg role="img" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
+            <path fill="currentColor" d="M9.41,8l5.29-5.29a1,1,0,0,0-1.41-1.41L8,6.59,2.71,1.29A1,1,0,0,0,1.29,2.71L6.59,8,1.29,13.29a1,1,0,1,0,1.41,1.41L8,9.41l5.29,5.29a1,1,0,0,0,1.41-1.41Z"/>
+        </svg>
+    </Button>
+);
+
+class Modal extends React.PureComponent<ModalProps & React.HTMLAttributes<HTMLDivElement>> {
+
+    public static propTypes = {
+        children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
+        className: PropTypes.string,
+        id: PropTypes.string,
+        isOpen: PropTypes.bool,
+        onDismiss: PropTypes.func,
+        title: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node, PropTypes.string]).isRequired
+    };
```

Comment:
Just curious, but John mentioned the other day that he hoped this would be generated in the transpiled code without it being specified.  Did we happen to check into that at all?

Reply:
I'm not sure this is something we'll get for free from TS. Investigating our options now. Hopefully there's a library or something that can autogenerate propTypes given some interface.